src/gen_rules.ml

```diff
@@ -63,8 +63,18 @@ module Gen(P : sig val sctx : Super_context.t end) = struct
   let with_format sctx ~dir ~f =
     let scope = SC.find_scope_by_dir sctx dir in
     let project = Scope.project scope in
-    Dune_project.find_extension_args project Auto_format.key
-    |> Option.iter ~f
+    let from_ext =
+      Dune_project.find_extension_args project Format_extension.key
+    in
+    let from_dune_lang = Dune_project.format_config project in
```

Comment:
Have you considered doing this merging in `dune_project.ml` directly? It seems better to merge the two fields as soon as possible. Then the key could stay a detail hidden inside `dune_project.ml`.

Reply:
Good idea, just did that.